esphome/components/font/__init__.py

```diff
@@ -142,25 +134,48 @@ def validate_weight_name(value):
     return FONT_WEIGHTS[cv.one_of(*FONT_WEIGHTS, lower=True, space="-")(value)]
 
 
-def download_gfonts(value):
-    wght = value[CONF_WEIGHT]
-    if value[CONF_ITALIC]:
-        wght = f"1,{wght}"
-    name = f"{value[CONF_FAMILY]}@{value[CONF_WEIGHT]}"
-    url = f"https://fonts.googleapis.com/css2?family={value[CONF_FAMILY]}:wght@{wght}"
+def get_font_url(value):
+    if value[CONF_TYPE] == TYPE_GFONTS:
+        wght = value[CONF_WEIGHT]
+        return (
+            f"https://fonts.googleapis.com/css2?family={value[CONF_FAMILY]}:wght@{wght}"
+        )
+    if value[CONF_TYPE] == TYPE_WEB:
+        return value[CONF_URL]
+    return ""
```

Comment:
Maybe better return None, instead of an empty string?
```suggestion
    return None
```